File: array.c
// // 747. Largest Number At Least Twice of Others

// class Solution {
// public:
//     int dominantIndex(vector<int>& nums) {
//         int first = 0,second = 0;
//         int index = 0;
//         int arr_len = nums.size();
        
//         for(int i=0;i<arr_len;i++){
//             if(nums[i]>first){
//                 second = first;
//                 first = nums[i];
//                 index = i;
//             }
//             else if(nums[i]>second){
//                 second = nums[i];
//             }
//         }
        
//         if(first>=2*second){
//             return index;
//         }
//         else{
//             return -1;
//         }
//     }
// };

// // 1. Two Sum
// // 返回target
// class Solution {
// public:
//     vector<int> twoSum(vector<int>& nums, int target) {
        
//         vector<int> result;
//         vector<int> array;
//         array.assign(nums.begin(), nums.end());
//         sort(array.begin(),array.end());
//         int current_left = 0;
//         int current_right = (int)nums.size()-1;
//         while (current_left<current_right) {
//             if(array[current_left]+array[current_right]==target){
// //                int index_1 = (int)(find(nums.begin(), nums.end(), array[current_left]) - nums.begin());
// //                int index_2 = (int)(find(nums.begin(), nums.end(), array[current_right]) - nums.begin());
//                 result.push_back(array[current_left]);
//                 result.push_back(array[current_right]);
//                 break;
//             }
//             else if(array[current_left]+array[current_right]<target){
//                 current_left++;
//             }
//             else{
//                 current_right--;
//             }
//         }
//         return result;
//     }
// };

// class Solution {
// public:
//     vector<int> twoSum(vector<int>& nums, int target) {
//         unordered_map<int,int>map;
//         vector<int> res;
//         for(int i=0;i<nums.size();i++){
//             int num2find = target - nums[i];
//             if(map.find(num2find)!=map.end()){
//                 res.push_back(map[num2find]);
//                 res.push_back(i);
//             }
//             map[nums[i]] = i;
//         }
//         return res;
//     }
// };



// // Merge Sorted Array
// class Solution {
// public:
//     void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
//         if(n==0){
//             return;
//         }
//         if(m==0){
//             nums1.assign(nums2.begin(), nums2.end());
//         }
//         int idx1=m-1,idx2=n-1,idx_glo=m+n-1;
//         while (idx1>=0 && idx2>=0) {
//             if(nums1[idx1]>nums2[idx2]){
//                 nums1[idx_glo--]=nums1[idx1--];
//             }
//             else{
//                 nums1[idx_glo--]=nums2[idx2--];
//             }
//         }
//         while (idx2>=0) {
//             nums1[idx_glo--]=nums2[idx2--];
//         }
//     }
// };   

//  746. Min Cost Climbing Stairs

// class Solution {
// public:
//     int minCostClimbingStairs(vector<int>& cost) {
//         int dp[cost.size()];
//         dp[0]=cost[0];
//         dp[1]=cost[1];
//         for(int i=2;i<cost.size();i++){
//             dp[i]=min(dp[i-1], dp[i-2])+cost[i];
//         }
//         return min(dp[cost.size()-2], dp[cost.size()-1]);
//     }
// private:
//     int min(int a, int b){
//         return a>b ? b:a;
//     }
// };


// // 724. Find Pivot Index
// class Solution {
// public:
//     int pivotIndex(vector<int>& nums) {
//         if(nums.size()<=2){
//             return -1;
//         }
//         int sum1[nums.size()];
//         sum1[0] = nums[0];
//         for(int i=1;i<nums.size();i++){
//             sum1[i]=sum1[i-1]+nums[i];
//         }
//         int left_sum = 0;
//         for(int i=0;i<nums.size();i++){
//             if(sum1[i]==sum1[nums.size()-1]-left_sum){
//                 return i;
//             }
//             else{
//                 left_sum = sum1[i];
//             }
//             printf("%d ",left_sum);
            
//         }
//         return -1;
//     }
// };

// // 717. 1-bit and 2-bit Characters
// class Solution {
// public:
//     bool isOneBitCharacter(vector<int>& bits) {
//         int n = (int)bits.size();
//         int i = 0;
//         while (i<n-1) {
//             if(bits[i]==0){
//                 i++;
//             }
//             else{
//                 i+=2;
//             }
//         }
//         return i==n-1;
//     }
// };

// // 26. Remove Duplicates from Sorted Array

// class Solution {
// public:
//     int removeDuplicates(vector<int>& nums) {
//         if(nums.size()==0){
//             return 0;
//         }
//         int current_index=0;
//         int dif_index = 1;
//         for(;dif_index<nums.size();dif_index++){
//             if(nums[dif_index]!=nums[current_index]){
//                 nums[++current_index] = nums[dif_index];
//             }
//         }
//         return current_index+1;
//     }
// };

// // 665. Non-decreasing Array

// class Solution {
// public:
//     bool checkPossibility(vector<int>& nums) {
//         int len = (int)nums.size();
//         bool decrease = false;
//         if(len<=1) return true;
//         for(int i=1;i<len;i++){
//             if(nums[i]<nums[i-1]){
//                 if(decrease){
//                     return false;
//                 }
//                 else{
//                     if(i-2>=0 && nums[i]<nums[i-2]){
//                         nums[i]=nums[i-1];
//                     }
//                     decrease = true;
//                 }
//             }
//         }
//         return true;
//     }
// };

// // 766. Toeplitz Matrix

// class Solution {
// public:
//     bool isToeplitzMatrix(vector<vector<int>>& matrix) {
//         int row = (int)matrix.size();
//         int col = (int)matrix[0].size();
// //        行扫描
//         for(int i=0;i<col;i++){
//             int this_element = matrix[0][i];
//             for(int m=0,n=i;m<row && n<col; m++,n++){
//                 if(matrix[m][n] != this_element){
//                     return false;
//                 }
//             }
//         }
// // 列扫描        
//         for(int j=0;j<row;j++){
//             int this_element = matrix[j][0];
//             for(int m=j,n=0;m<row && n<col; m++,n++){
//                 if(matrix[m][n] != this_element){
//                     return false;
//                 }
//             }
//         }
        
//         return true;
//     }
// };


// 主要是看的哈希值


// // 697. Degree of an Array


// class Solution {
// public:
    
// //     int max(int a,int b){
// //         return a>b ? a:b;
// //     }
    
// //     int min(int a,int b){
// //         return a<b ? a:b;
// //     }
    
//     int findShortestSubArray(vector<int>& nums) {
//         unordered_map<int, int>num_fre;
//         unordered_map<int, int>start_index;
//         unordered_map<int, int>end_index;
//         unordered_map<int, int >::iterator iter;
//         int max_fre = INT_MIN;
//         for(int i=0;i<nums.size();i++){
//             int num = nums[i];
//             if (start_index.find(num)==start_index.end()){
//                 num_fre[num]=0;
//                 start_index[num]=i;
//             }
//             num_fre[num] += 1;
//             end_index[num]=i;
//             max_fre = max(max_fre,num_fre[num]);
//         }
        
//         int res = INT_MAX;
        
//         for(iter = num_fre.begin();iter!=num_fre.end();iter++){
//             if(iter->second == max_fre){
//                 res = min(res,end_index[iter->first] - start_index[iter->first]+1);
//             }
//         }
//         return res;
//     }
// };


// 674. Longest Continuous Increasing Subsequence

// class Solution {
// public:
//     int findLengthOfLCIS(vector<int>& nums) {
//         if(nums.size()==0) return 0;
//         int res=1;
//         int current_LCIS = 1;
//         for(int i=0;i<nums.size()-1;i++){
//             if (nums[i+1]>nums[i]) {
//                 current_LCIS+=1;
//             }
//             else{
//                 current_LCIS = 1;
//             }
//             res = max(current_LCIS,res);
//         }
//         return res;
//     }
// };


// // 661. Image Smoother

// class Solution {
// public:
    
//     vector<vector<int>> imageSmoother(vector<vector<int>>& M) {
//         vector<vector<int>> res;
// // vector<vector<int>> res(row ,vector<int>(col));
//         res.assign(M.begin(), M.end());
//         int row = (int)M.size();
//         int col = (int)M[0].size();
//         for(int i =0;i<row;i++){
//             for(int j=0;j<col;j++){
//                 int sum = 0;
//                 int count = 0;
//                 for(int m=-1;m<=1;m++){
//                     for(int n=-1;n<=1;n++){
//                         int x_axis = m+i;
//                         int y_axis = n+j;
//                         if(x_axis>=0 && x_axis<row && y_axis>=0 && y_axis<col){
//                             sum+=M[x_axis][y_axis];
//                             count+=1;
//                         }
//                     }
//                 }
//                 res[i][j] = sum/count;
//             }
//         }
//         return res;
//     }
// };


// // 643. Maximum Average Subarray I

// class Solution {
// public:
    
//     double max(double a, double b){
//         return a>b?a:b;
//     }
//     double findMaxAverage(vector<int>& nums, int k) {
//         int window_sum = 0;
//         for(int i=0;i<k;i++){
//             window_sum+=nums[i];
//         }
//         double maxAverage = window_sum/(k*1.0);
//         for(int i=0;i<nums.size()-k;i++){
//             window_sum = window_sum + nums[i+k] - nums[i];
//             double average = window_sum/(k*1.0);
//             maxAverage = max(maxAverage,average);
//         }
//         return maxAverage;
//     }
// };

// // 628. Maximum Product of Three Numbers

// class Solution {
// public:
//     static bool cmp(int a,int b){
//         return a>b;
//     }
    
//     int maximumProduct(vector<int>& nums) {
//         sort(nums.begin(), nums.end(),cmp);
//         int ans1 = nums[0]*nums[1]*nums[2];
//         int ans2 = nums[nums.size()-1]*nums[nums.size()-2]*nums[0];
//         return max(ans1, ans2);
//     }
// };

// // 605. Can Place Flowers
// class Solution {
// public:
//     bool canPlaceFlowers(vector<int>& flowerbed, int n) {
//         int count=0;
//         for(int i=0;i<flowerbed.size();i++){
//             if(i==0){
//                 if(flowerbed[i]==0 && flowerbed[i+1]==0) {
//                     count++;
//                     flowerbed[i]=1;
//                 }
//             }
//             else if (i==flowerbed.size()-1){
//                 if(flowerbed[i]==0 && flowerbed[i-1]==0){
//                     count++;
//                     flowerbed[i]=1;
//                 }
//             }
//             else if(flowerbed[i-1]==0 && flowerbed[i]==0 && flowerbed[i+1]==0){ flowerbed[i]=1;
//                 count++;
//             }
//         }
//         printf("%d",count);
//         return count>=n;
//     }
// };


// // 566. Reshape the Matrix

// class Solution {
// public:
//     vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
//         int original_row = (int)nums.size();
//         int original_col = (int)nums[0].size();
//         if(r*c != original_col*original_row) return nums;
//         vector<vector<int>>res(r,(vector<int>(c)));
//         int order =0;
//         for(int i=0;i<r;i++){
//             for(int j=0;j<c;j++){
//                 int m = order/original_col;
//                 int n = order%original_col;
//                 // printf("%d,%d,%d\n",order,m,n);
//                 res[i][j]=nums[m][n];
//                 order++;
//             }
//         }
//         return res;
//     }
// };

// // 561. Array Partition I

// class Solution {
// public:
//     int arrayPairSum(vector<int>& nums) {
//         int sum = 0;
//         sort(nums.begin(), nums.end());
//         for(int i=0;i<nums.size();i+=2){
//             sum += nums[i];
//         }
//         return sum;
//     }
// };

// // 485. Max Consecutive Ones
// class Solution {
// public:
//     int findMaxConsecutiveOnes(vector<int>& nums) {
//         int max_consecutive = 0;
//         int cur_consecutive = 0;
//         for(int i=0;i<nums.size();i++){
//             if(nums[i]==1){
//                 cur_consecutive+=1;
//                 max_consecutive = max(max_consecutive,cur_consecutive);
//             }
//             else{
//                 cur_consecutive = 0;
//             }
//         }
//         return max_consecutive;
//     }
// };

// // 448. Find All Numbers Disappeared in an Array

// class Solution {
// public:
    
//     void swap(int *a, int *b){
//         int temp = *a;
//         *a = *b;
//         *b = temp;
//     }
    
//     vector<int> findDisappearedNumbers(vector<int>& nums) {
//         for(int i=0;i<nums.size();i++){
// //        使每个位置的元素都在自己的位置上，直到该元素的位置上就是该元素
//             while (nums[i]!=i+1 && nums[nums[i]-1]!=nums[i]) {
//                 swap(&nums[i], &nums[nums[i]-1]);
//             }
//         }
//         vector<int> res;
//         for(int i=0;i<nums.size();i++){
//             if(nums[i]!=i+1){
//                 res.push_back(i+1);
//             }
//         }
//         return res;
//     }
// };

// // # 283. Move Zeroes

// class Solution {
// public:
//     void moveZeroes(vector<int>& nums) {
//         int current_pos = 0;
//         for(int i=0;i<nums.size();i++){
//             if(nums[i]!=0){
//                 nums[current_pos++]=nums[i];
//             }
//         }
//         for(int i=current_pos;i<nums.size();i++){
//             nums[i]=0;
//         }
//     }
// };

// // # 217. Contains Duplicate
// class Solution {
// public:
//     bool containsDuplicate(vector<int>& nums) {
//         int numsSize = (int)nums.size();
//         if(numsSize<=1){
//             return false;
//         }
//         unordered_map<int, int>map;
//         for(int i=0;i<numsSize;i++){
//             if(map.find(nums[i])!=map.end()){
//                 return true;
//             }
//             else{
//                 map[nums[i]]=1;
//             }
//         }
//         return false;
//     }
// };

// // 268. Missing Number
// class Solution {
// public:
//     int missingNumber(vector<int>& nums) {
//         bool exist_max = false;
//         for(int i=0;i<nums.size();i++){
//             if(abs(nums[i])==nums.size()) exist_max=true;
//             int index = abs(nums[i]);
//             nums[index] = -nums[index];
//         }
//         for(int i=0;i<nums.size();i++){
//             if(nums[i]>0){
//                 return i;
//             }
//         }
//         if(exist_max){
//             for(int i=0;i<nums.size();i++){
//                 if(nums[i]>=0){
//                     return i;
//                 }
//             }
//         }
//         return (int)nums.size();
//     }
// };

// // 169. Majority Element

// class Solution {
// public:
//     int majorityElement(vector<int>& nums) {
//         unordered_map<int, int>map;
//         for(int i=0;i<nums.size();i++){
//             if(map.find(nums[i])!=map.end()){
//                 map[nums[i]]+=1;
//             }
//             else{
//                 map[nums[i]]=1;
//             }
//         }
//         unordered_map<int, int>::iterator iter;
//         for(iter=map.begin();iter!=map.end();iter++){
//             if(iter->second>nums.size()/2){
//                 return iter->first;
//             }
//         }
//         return 0;
//     }
// };

// // 167. Two Sum II - Input array is sorted

// class Solution {
// public:
//     vector<int> twoSum(vector<int>& numbers, int target) {
//         unordered_map<int, int>hash;
//         vector<int>res;
//         for(int i=0;i<numbers.size();i++){
//             int num2find = target-numbers[i];
//             if(hash.find(num2find)!=hash.end()){
//                 res.push_back(hash[num2find]+1);
//                 res.push_back(i+1);
//             }
//             else{
//                 hash[numbers[i]]=i;
//             }
//         }
//         return res;
//     }
// };

// // 122. Best Time to Buy and Sell Stock II

// class Solution {
// public:
//     int maxProfit(vector<int>& prices) {
//         if(prices.size()<=1) return 0;
//         int max_profit=0;
//         for(int i=0;i<prices.size()-1;i++){
//             if(prices[i+1]>prices[i]){
//                 max_profit += prices[i+1]-prices[i];
//             }
//         }
//         return max_profit;
//     }
// };

File: array.py
# # 747. Largest Number At Least Twice of Others
# class Solution:
#     def dominantIndex(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         from copy import deepcopy

#         arry = deepcopy(nums)

#         if len(arry) == 0:
#             return -1
#         elif len(arry) == 1:
#             return 0
#         else:
#             arry.sort()
#             if arry[-1] >= 2*arry[-2]:
#                 return nums.index(arry[-1])
#             else:
#                 return -1

# # 1. Two Sum
# class Solution:
#     def twoSum(self, nums, target):
#         """
#         :type nums: List[int]
#         :type target: int
#         :rtype: List[int]
#         """
#         nummap = {}
#         for index in range(0, len(nums)):
#             num2find = target - nums[index]
#             if num2find in nummap:
#                 return nummap[num2find], index
#             else:
#                 nummap[nums[index]] = index


# # Merge Sorted Array
# class Solution:
#     def merge(self, nums1, m, nums2, n):
#         idx1 = m-1
#         idx2 = n-1
#         idx_glo = m+n-1
#         while idx1 >= 0 and idx2 >= 0:
#             if nums1[idx1] > nums2[idx2]:
#                 nums1[idx_glo] = nums1[idx1]
#                 idx1 -= 1
#                 idx_glo -= 1
#             else:
#                 nums1[idx_glo] = nums2[idx2]
#                 idx2 -= 1
#                 idx_glo -= 1
#         while idx2 >= 0:
#             nums1[idx_glo] = nums2[idx2]
#             idx2 -= 1
#             idx_glo -= 1


# # 746. Min Cost Climbing Stairs

# class Solution:
#     def minCostClimbingStairs(self, cost):
#         """
#         :type cost: List[int]
#         :rtype: int
#         """
#         size = len(cost)
#         dp = [cost[0], cost[1]]
#         for x in range(2, size):
#             dp.append(min(dp[x-1], dp[x-2])+cost[x])
#         return min(dp[-1], dp[-2]) 


# # 724. Find Pivot Index

# class Solution:
#     def pivotIndex(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         from copy import deepcopy
#         if len(nums) <= 2:
#             return -1
#         sum1 = deepcopy(nums)
#         sum2 = deepcopy(nums)
#         for i in range(1, len(nums)):
#             sum1[i] = sum1[i-1] + sum1[i]
#         for i in reversed(range(0, len(nums)-1)):
#             sum2[i] = sum2[i] + sum2[i+1]
#         print(sum1)
#         print(sum2)
#         for i in range(0, len(nums)):
#             if sum1[i] == sum2[i]:
#                 return i

#         return -1


# 717. 1-bit and 2-bit Characters

# class Solution:
#     def isOneBitCharacter(self, bits):
#         """
#         :type bits: List[int]
#         :rtype: bool
#         """
#         arr_len = len(bits)
#         i = 0
#         while i < arr_len-1:
#             i += bits[i] + 1

#         return i == arr_len-1

# 26. Remove Duplicates from Sorted Array

# class Solution:
#     def removeDuplicates(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         if len(nums) == 0:
#             return 0
#         current_index = 0
#         for diff_index in range(1, len(nums)):
#             if nums[current_index] != nums[diff_index]:
#                 current_index += 1
#                 nums[current_index] = nums[diff_index]
#         return current_index + 1


# # 665. Non-decreasing Array

# class Solution:
#     def checkPossibility(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: bool
#         """
#         decrease = False
#         if len(nums) <= 1:
#             return True
#         for i in range(1, len(nums)):
#             if nums[i] < nums[i-1]:
#                 if decrease:
#                     return False
#                 else:
#                     if i-2 >= 0 and nums[i] < nums[i-2]:
#                         nums[i] = nums[i-1]
#                     decrease = True
#         return True


# # 697. Degree of an Array

# class Solution:
#     def findShortestSubArray(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         import sys
#         num_fre = {}
#         sub_start = {}
#         sub_end = {}
#         max_fre = -sys.maxsize
#         for i, num in enumerate(nums):
#             # 如果在的话
#             if num not in sub_start.keys():
#                 num_fre[num] = 0
#                 sub_start[num] = i

#             num_fre[num] += 1
#             sub_end[num] = i
#             max_fre = max(max_fre, num_fre[num])
#         res = sys.maxsize
#         for num, count in num_fre.items():
#             if count == max_fre:
#                 res = min(res, sub_end[num]-sub_start[num]+1)
#         return res


# # 695. Max Area of Island
# class Solution:
#     def maxAreaOfIsland(self, grid):
#         """
#         :type grid: List[List[int]]
#         :rtype: int
#         """
#         maxArea = 0
#         for i in range(0, len(grid)):
#             for j in range(0, len(grid[0])):
#                 if grid[i][j] != 0:
#                     maxArea = max(maxArea, self.areaOfIsland(grid, i, j))
#         return maxArea

#     def areaOfIsland(self, grid, i, j):
#         if i>=0 and i < len(grid) and j>=0 and j< len(grid[0]) and grid[i][j]!=0:
#             grid[i][j] = 0
#             return 1 + self.areaOfIsland(grid, i-1, j) + self.areaOfIsland(grid, i+1, j) + self.areaOfIsland(grid, i, j-1) + self.areaOfIsland(grid, i, j+1)
#         return 0


# # 674. Longest Continuous Increasing Subsequence
# class Solution:
#     def findLengthOfLCIS(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         if len(nums) == 0:
#             return 0

#         res = 1
#         cursor = 1

#         for i in range(0,len(nums)-1):
#             if nums[i] < nums[i+1]:
#                 cursor += 1
#             else:
#                 cursor = 1
#             res = max(res, cursor)

#         return res


# # 661. Image Smoother

# class Solution:
#     def imageSmoother(self, M):
#         """
#         :type M: List[List[int]]
#         :rtype: List[List[int]]
#         """
#         # from copy import deepcopy
#         # res = deepcopy(M)
#         col = len(M[0])
#         row = len(M)
#         # res = [[0]*col]*row
#         res = [None]*row
#         for i in range(0, row):
#             res[i] = [0]*col
#         for i in range(0, row):
#             for j in range(0, col):
#                 count_sum = 0
#                 count = 0
#                 for m in range(-1,2):
#                     for n in range(-1,2):
#                         x = i+m
#                         y = j+n
#                         if 0 <= x < len(M) and 0 <= y < len(M[0]):
#                             # print(x,y)
#                             count_sum += M[x][y]
#                             count += 1
#                 res[i][j] = count_sum//count
#         return res


# 643. Maximum Average Subarray I

# class Solution:
    # def findMaxAverage(self, nums, k):
    #     """
    #     :type nums: List[int]
    #     :type k: int
    #     :rtype: float
    #     """
    #     if k == len(nums):
    #         return sum(nums) / k
    #     window_sum = sum(nums[0:k])
    #     maxAverage = window_sum / k
    #     for i in range(0, len(nums) - k):
    #         window_sum = window_sum + nums[i+k] - nums[i]
    #         average = window_sum / k
    #         maxAverage = max(maxAverage, average)
    #     return maxAverage


# # 628. Maximum Product of Three Numbers

# class Solution(object):
# def maximumProduct(self, nums):
#     """
#     :type nums: List[int]
#     :rtype: int
#     """
#     sorted_num = sorted(nums)
#     sorted_num.reverse()
#     ans1 = sorted_num[0]*sorted_num[1]*sorted_num[2]
#     ans2 = sorted_num[-1]*sorted_num[-2]*sorted_num[0]
#     return max(ans1, ans2)

# # 605. Can Place Flowers
# class Solution:
#     def canPlaceFlowers(self, flowerbed, n):
#         """
#         :type flowerbed: List[int]
#         :type n: int
#         :rtype: bool
#         """
#         count = 0

#         if len(flowerbed) == 1:
#             if flowerbed[0] == 0:
#                 count += 1
#             return count >= n

#         for i in range(0, len(flowerbed)):
#             if i == 0:
#                 if flowerbed[i] == 0 and flowerbed[i+1] == 0:
#                     count += 1
#                     flowerbed[i] = 1
#             elif i == len(flowerbed)-1:
#                 if flowerbed[i] == 0 and flowerbed[i-1] == 0:
#                     count += 1
#                     flowerbed[i] = 1
#             elif flowerbed[i-1] == 0 and flowerbed[i] == 0 and flowerbed[i+1] == 0:
#                 count += 1
#                 flowerbed[i] = 1
#         return count >= n

# # 566. Reshape the Matrix

# class Solution:
#     def matrixReshape(self, nums, r, c):
#         """
#         :type nums: List[List[int]]
#         :type r: int
#         :type c: int
#         :rtype: List[List[int]]
#         """
#         res = [None] * r
#         for i in range(0, r):
#             res[i] = [0] * c
#         original_row = len(nums)
#         original_col = len(nums[0])
#         if r * c != original_col * original_row: return nums
#         order = 0
#         for i in range(0, r):
#             for j in range(0, c):
#                 m = order // original_col
#                 n = order % original_col
#                 res[i][j] = nums[m][n]
#                 order += 1

#         return res

# # 448. Find All Numbers Disappeared in an Array

# class Solution:
#     def findDisappearedNumbers(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: List[int]
#         """
#         # 取负法，将有的元素所对应的位置的元素取负

#         for i in range(0, len(nums)):
#             index = abs(nums[i]) - 1
#             if nums[index] > 0:
#                 nums[index] = -nums[index]
        
#         res = []
#         for i in range(0, len(nums)):
#             if nums[i] > 0:
#                 res.append(i+1)
                
#         return res

# 283. Move Zeroes

# class Solution:
#     def moveZeroes(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: void Do not return anything, modify nums in-place instead.

#         """

#         # 是非零元素才动
#         j = 0
#         for i in range(0, len(nums)):
#             if nums[i] != 0:
#                 tmp = nums[i]
#                 nums[i] = nums[j]
#                 nums[j] = tmp
#                 j += 1




